Neos.ContentRepository.Core/Classes/Feature/WorkspaceRebase/CommandThatFailedDuringRebase.php

```diff
@@ -23,12 +23,12 @@
 final readonly class CommandThatFailedDuringRebase
 {
     /**
-     * @param CommandInterface $command the command that failed
+     * @param RebasableToOtherWorkspaceInterface $command the command that failed
      * @param \Throwable $exception how the command failed
      * @param SequenceNumber $sequenceNumber the event store sequence number of the event containing the command to be rebased
      */
     public function __construct(
-        public CommandInterface $command,
+        public RebasableToOtherWorkspaceInterface $command,
```

Comment:
IMHO that's the current problem of all this. All commands should have one common interface. public / private ones might be distinguished by another interface.

Also this whole thing here seems questionable, given that commands do not have anything in common and the interface declares absolutely no API they are a pretty useless set...

Reply:
jup its just a marker interface, for the neos ui we used this for conflict resolution see https://github.com/neos/neos-ui/pull/3881/files

basically try to find meaning in the command and say: if this was a set property then that failed on node blub and make it readable.

Not perfect but well ... the mean thing is we need to pass the low level commands here because we dont have the high level ones ... maybe it should just be marked as internal experimental extension point only for the ui ...

Reply:
I guess what I am saying is, if any third party wants to do _aynthing_ with a collection of events like this, we absoltuely should have an interface that declares some sensible api to work with. as it stands right now it almost doesn't matter if we give something internal to the UI or not because none of it brings any meaning. You could just as well stick a node in there, whatever is in there we would have to rummage in the internals of the object to see if it helps us. tl;dr; it seems a bad idea to put marker interfaces out to the world and all of them should be _internal_ ?

Reply:
coudl work yeah

Reply:
Adjusted see https://github.com/neos/neos-development-collection/pull/5348/commits/d79fe107f80ac4dbb779341aa28eebd70049628f and UI's https://github.com/neos/neos-ui/pull/3881/commits/8619c7729e815a590981840ab3e4ff60df300793